Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectILLReductionDIFF.py

```diff
@@ -0,0 +1,149 @@
+
+        self.declareProperty(MatrixWorkspaceProperty('OutputWorkspace', '', direction=Direction.Output),
+                             doc='The output workspace group containing reduced data.')
+
+        self.declareProperty("MaskPixelsFromStart", 10, validator=IntBoundedValidator(lower=0),
+                             doc="Number of pixels to mask at the start of each tube")
+        self.declareProperty("MaskPixelsFromEnd", 10, validator=IntBoundedValidator(lower=0),
+                             doc="Number of pixels to mask at the end of each tube")
+
+        self.declareProperty("Observable", "sample.temperature",
+                             doc="If multiple files, the parameter from SampleLog to use as an index when conjoined.")
+
+        self.declareProperty("Transpose", True, doc="Transpose the result.")
+
+    def _normalize_by_monitor(self):
+        """
+            Normalizes the workspace by monitor value (ID is 0 for IN16B)
+        """
+        monitor_ws = self.output + '_mon'
+        ExtractMonitors(InputWorkspace=self.output, DetectorWorkspace=self.output, MonitorWorkspace=monitor_ws)
+
+        Divide(LHSWorkspace=self.output, RHSWorkspace=monitor_ws, OutputWorkspace=self.output, WarnOnZeroDivide=True)
+
+        DeleteWorkspace(monitor_ws)
+
+    def _mask_tube_ends(self):
+        """
+            Mask the ends of each tube according to values provided by the user
+        """
+        # the numbers here correspond to IN16B detectors
+        cache = list(range(1, self.mask_start_pixels)) + list(range(257 - self.mask_end_pixels, 257))
+        to_mask = [i + 256 * j for i in cache for j in range(8)]
+        MaskDetectors(Workspace=self.output, DetectorList=to_mask)
+
+    def _treat_doppler(self, ws):
+        """
+            Reduce Doppler diffraction data presents in workspace.
+            @param ws: the input workspace
+        """
+        run = None
+        if len(self.runs) > 1:
+            run = mtd[ws][0].getRun()
+        else:
+            run = mtd[ws].getRun()
+
+        if run.hasProperty('Doppler.incident_energy'):
+            energy = run.getLogData('Doppler.incident_energy').value / 1000
+        else:
+            raise RuntimeError("Unable to find incident energy for Doppler mode")
+
+        Integration(InputWorkspace=ws, OutputWorkspace=self.output)
+        self._normalize_by_monitor()
+        self._mask_tube_ends()
+
+        ExtractUnmaskedSpectra(InputWorkspace=self.output, OutputWorkspace=self.output)
+
+        ConvertSpectrumAxis(InputWorkspace=self.output,
+                            OutputWorkspace=self.output,
+                            Target='ElasticQ',
+                            EMode="Direct",
+                            EFixed=energy)
+
+        ConvertToPointData(InputWorkspace=self.output, OutputWorkspace=self.output)
+
+        ConjoinXRuns(InputWorkspaces=self.output,
```

Comment:
It seems we don't have more data yet to test multiple files.
However from the code, it might be better to conjoin runs first.
Once conjoined, it will be a single workspace, so the rest of operations will be done once.
Whereas now, everything is done on per workspace basis, which is redundant.
Can you refactor to put conjoin right after integration?